Release: 0.16.1

ax3l opened the pull request:
- [x] See https://github.com/openPMD/openPMD-api/pull/1648#issuecomment-2398527616
- [x] Fix CMake superbuild control #1678
- [x] ULONG example fix #1710
- [x] Update macOS CI #1703

## Which version am I running?

### Python

```python
import openpmd_api
print(openpmd_api.__version__)
```

### C++

```C++
#include <openPMD/openPMD.hpp>
#include <iostream>

int main() {
    // query compile-time macros:
    std::cout << OPENPMDAPI_VERSION_MAJOR << "."
              << OPENPMDAPI_VERSION_MINOR << "."
              << OPENPMDAPI_VERSION_PATCH << std::endl;
    // there is also: OPENPMDAPI_VERSION_LABEL (e.g. "alpha")

    // or query the runtime API (since 0.12.0):
    std::cout << openPMD::getVersion() << std::endl;
    return 0;
}
```

### CLI

```bash
# command line option (since 0.12.0):
openpmd-ls --version
```

ax3l committed:
Release: 0.16.1

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>

franzpoeschel committed:
Fix new_version.py

franzpoeschel committed:
Update version in windows.yml

franzpoeschel committed:
Reformat with yapf3

franzpoeschel commented:
I have pushed a fixed version of the version script that now includes the missing files. The diff created by the script (when executed on the dev branch right before 0.16.0) is identical to this PR's diff.
There was one version update missing in the CI, pushed a fix there too.

franzpoeschel committed:
Fix greedy Regex parsing

ax3l commented:
Thank you @franzpoeschel ! :tada: 

Does the Pythonic suffix work as well?
- dev/rc `.` suffix: For `setup.py` it needs to read `x.y.zaN` and `x.y.z.dev` per PEP-440.
- dev/rc `-` suffix: In C++ and `docs/source/conf.py: "release"` (not `"version"`) we write `x.y.z` and `x.y.z-dev`.
- In `CMakeLists.txt` we use _no_ suffix for dev/rc versions.

ax3l commented on CHANGELOG.rst:
```diff
@@ -3,6 +3,15 @@
 
+0.16.1
+------
+**Date:** 2024-10-07
+
+New Backends, Extension, Perf. & Memory
+
+This is the 0.16.0 release but with internal version bumps where they were missing.
```
To update once all fixes are in.

franzpoeschel replied:
Are any further fixes needed?
We could bump toml11 back to 4.2 now that we know the segfault was my setup's fault.

ax3l replied:
I am ok to generally bump to 4.2.0+ now, because toml11 has a follow-up release (4.3.0) by now and we are not bound to a single version that works anymore.

franzpoeschel replied:
Yep, it's part of the release. This PR contains commit https://github.com/openPMD/openPMD-api/pull/1676/commits/987018e3b268bad53bb86de0a7a4f51c9a9e91db that sets 4.2 as the default version again. Otherwise, 4.3 if people include it as an external version.

ax3l committed:
Fix: CMake Internal Control (#1678)

The `openPMD_USE_INTERNAL_*` options are for general internal/external
control. They were accidentally overwritten by new tar flags
and sometimes incomplete.

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>

franzpoeschel commented:
> Thank you @franzpoeschel ! 🎉
> 
> Does the Pythonic suffix work as well?
> 
>     * dev/rc `.` suffix: For `setup.py` it needs to read `x.y.zaN` and `x.y.z.dev` per PEP-440.
> 
>     * dev/rc `-` suffix: In C++ and `docs/source/conf.py: "release"` (not `"version"`) we write `x.y.z` and `x.y.z-dev`.
> 
>     * In `CMakeLists.txt` we use _no_ suffix for dev/rc versions.

I tried paying atttention to this. This is the diff after using the script to enable version 0.17.0-dev:
```diff
diff --git a/.github/workflows/windows.yml b/.github/workflows/windows.yml
index 183bcc4f7..5d132a622 100644
--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -42,7 +42,7 @@ jobs:
 
         python3.exe -m pip wheel .
         if(!$?) { Exit $LASTEXITCODE }
-        python3.exe -m pip install openPMD_api-0.16.1-cp39-cp39-win_amd64.whl
+        python3.exe -m pip install openPMD_api-0.17.0.dev0-cp39-cp39-win_amd64.whl
         if(!$?) { Exit $LASTEXITCODE }
 
         python3.exe -c "import openpmd_api as api; print(api.variants)"
diff --git a/CITATION.cff b/CITATION.cff
index a0f24d52e..9504dca15 100644
--- a/CITATION.cff
+++ b/CITATION.cff
@@ -37,7 +37,7 @@ contact:
   orcid: https://orcid.org/0000-0003-1943-7141
   email: axelhuebl@lbl.gov
 title: "openPMD-api: C++ & Python API for Scientific I/O with openPMD"
-version: 0.16.1
+version: 0.17.0-dev
 repository-code: https://github.com/openPMD/openPMD-api
 doi: 10.14278/rodare.27
 license: LGPL-3.0-or-later
diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf4403671..fc5d6d5de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 #
 cmake_minimum_required(VERSION 3.22.0)
 
-project(openPMD VERSION 0.16.1) # LANGUAGES CXX
+project(openPMD VERSION 0.17.0) # LANGUAGES CXX
 
 # the openPMD "markup"/"schema" standard version
 set(openPMD_STANDARD_VERSION 1.1.0)
diff --git a/README.md b/README.md
index 0791a4fea..57fdc2150 100644
--- a/README.md
+++ b/README.md
@@ -304,7 +304,7 @@ export CMAKE_PREFIX_PATH=$HOME/somepath:$CMAKE_PREFIX_PATH
 Use the following lines in your project's `CMakeLists.txt`:
 ´´´cmake
 # supports:                       COMPONENTS MPI NOMPI HDF5 ADIOS2
-find_package(openPMD 0.16.1 CONFIG)
+find_package(openPMD 0.17.0 CONFIG)
 
 if(openPMD_FOUND)
     target_link_libraries(YourTarget PRIVATE openPMD::openPMD)
@@ -332,7 +332,7 @@ set(openPMD_INSTALL OFF)            # or instead use:
 set(openPMD_USE_PYTHON OFF)
 FetchContent_Declare(openPMD
   GIT_REPOSITORY "https://github.com/openPMD/openPMD-api.git"
-  GIT_TAG        "0.16.1")
+  GIT_TAG        "0.17.0")
 FetchContent_MakeAvailable(openPMD)
 ´´´
 
diff --git a/docs/source/conf.py b/docs/source/conf.py
index f93e0a276..af1253a13 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -86,9 +86,9 @@ author = u'The openPMD Community'
 # built documents.
 #
 # The short X.Y version.
-version = u'0.16.1'
+version = u'0.17.0'
 # The full version, including alpha/beta/rc tags.
-release = u'0.16.1'
+release = u'0.17.0-dev'
 
 # The language for content autogenerated by Sphinx. Refer to documentation
 # for a list of supported languages.
diff --git a/docs/source/dev/linking.rst b/docs/source/dev/linking.rst
index 5048ddbf1..033f81383 100644
--- a/docs/source/dev/linking.rst
+++ b/docs/source/dev/linking.rst
@@ -23,7 +23,7 @@ Use the following lines in your project's ``CMakeLists.txt``:
 .. code-block:: cmake
 
    # supports:                       COMPONENTS MPI NOMPI HDF5 ADIOS2
-   find_package(openPMD 0.16.1 CONFIG)
+   find_package(openPMD 0.17.0 CONFIG)
 
    if(openPMD_FOUND)
        target_link_libraries(YourTarget PRIVATE openPMD::openPMD)
@@ -53,7 +53,7 @@ Just replace the ``add_subdirectory`` call with:
    set(openPMD_USE_PYTHON OFF)
    FetchContent_Declare(openPMD
      GIT_REPOSITORY "https://github.com/openPMD/openPMD-api.git"
-     GIT_TAG        "0.16.1")
+     GIT_TAG        "0.17.0")
    FetchContent_MakeAvailable(openPMD)
 
 
diff --git a/docs/source/index.rst b/docs/source/index.rst
index e38b87d48..db2bbc300 100644
--- a/docs/source/index.rst
+++ b/docs/source/index.rst
@@ -44,7 +44,7 @@ openPMD-api version      supported openPMD standard versions
 ======================== ===================================
 ``2.0.0+``               ``2.0.0+``      (not released yet)
 ``1.0.0+``               ``1.0.1-1.1.0`` (not released yet)
-``0.13.1-0.16.1`` (beta) ``1.0.0-1.1.0``
+``0.13.1-0.17.0`` (beta) ``1.0.0-1.1.0``
 ``0.1.0-0.12.0`` (alpha) ``1.0.0-1.1.0``
 ======================== ===================================
 
diff --git a/include/openPMD/version.hpp b/include/openPMD/version.hpp
index ead912a23..bc1de8a5c 100644
--- a/include/openPMD/version.hpp
+++ b/include/openPMD/version.hpp
@@ -28,9 +28,9 @@
  * @{
  */
 #define OPENPMDAPI_VERSION_MAJOR 0
-#define OPENPMDAPI_VERSION_MINOR 16
-#define OPENPMDAPI_VERSION_PATCH 1
-#define OPENPMDAPI_VERSION_LABEL ""
+#define OPENPMDAPI_VERSION_MINOR 17
+#define OPENPMDAPI_VERSION_PATCH 0
+#define OPENPMDAPI_VERSION_LABEL "dev"
 /** @} */
 
 /** maximum supported version of the openPMD standard (read & write,
diff --git a/setup.py b/setup.py
index 271f24bc5..f1b88f23a 100644
--- a/setup.py
+++ b/setup.py
@@ -174,7 +174,7 @@ with open('./requirements.txt') as f:
 setup(
     name='openPMD-api',
     # note PEP-440 syntax: x.y.zaN but x.y.z.devN
-    version='0.16.1',
+    version='0.17.0.dev',
     author='Axel Huebl, Franz Poeschel, Fabian Koller, Junmin Gu',
     author_email='axelhuebl@lbl.gov, f.poeschel@hzdr.de',
     maintainer='Axel Huebl',
diff --git a/test/SerialIOTest.cpp b/test/SerialIOTest.cpp
index 439d17c32..b21e45927 100644
--- a/test/SerialIOTest.cpp
+++ b/test/SerialIOTest.cpp
@@ -6363,7 +6363,7 @@ void adios2_bp5_no_steps(bool usesteps)
         IO.DefineAttribute("/openPMD", std::string("1.1.0"));
         IO.DefineAttribute("/openPMDextension", uint32_t(0));
         IO.DefineAttribute("/software", std::string("openPMD-api"));
-        IO.DefineAttribute("/softwareVersion", std::string("0.16.1"));
+        IO.DefineAttribute("/softwareVersion", std::string("0.17.0-dev"));
 
         IO.DefineAttribute("/data/0/dt", double(1));
         IO.DefineAttribute(
```

franzpoeschel committed:
Revert "CMake Tar: TOML11 3.7.1 (#1673)" (#1679)

This reverts commit aea4b72388351e89214c0c3c3246acde2375e3ec.

franzpoeschel committed:
Replace deprecated makeSuite with loadTestsFromTestCase (#1681)

Removed with Python 3.13

franzpoeschel committed:
Changelog: Add PRs #1681 #1679 #1678

ax3l committed:
Doc: Fix Shipped Internally (#1682)

Downloaded now.

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>

ax3l committed:
`PYBIND11_FINDPYTHON=ON` (#1684)

Reuse our `find_package(Python ...)` call and use new CMake logic in
pybind11.
https://pybind11.readthedocs.io/en/stable/compiling.html#modules-with-cmake
https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-version-selection

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>

franzpoeschel committed:
Fix ICX build (#1690)

Error was:
 /home/runner/work/openPMD-api/openPMD-api/include/openPMD/backend/Container.hpp:262:32:
error: no member named 'm_container' in 'Container<T, T_key,
T_container>'
  262 |         container().swap(other.m_container);

ax3l committed:
Fix Leftover ADIOS1 Mentions (#1683)

* Fix Leftover ADIOS1 Mentions

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>

* Remove ADIOS1 from examples

---------

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>
Co-authored-by: Franz Pöschel <franz.poeschel@gmail.com>

franzpoeschel committed:
Explicitly specify ADIOS2 components in openPMDConfig.cmake (#1693)

* Explicitly specify ADIOS2 components in openPMDConfig.cmake

When including openPMD in downstream CMake and that CMake script
contains Fortran, ADIOS2 will automatically detect that. But since we
do not reexport the ADIOS2 headers, we only need the components that
openPMD explicitly uses: CXX and MPI.

* Same in CMakeLists.txt

franzpoeschel committed:
Remove mpirun_workaround.sh (#1698)

Seems to no longer be necessary in the affected CI runs

franzpoeschel committed:
Upgrade appleclang14 build to MacOS 13

> The macOS-12 environment is deprecated, consider switching to macOS-13,
macOS-14 (macos-latest) or macOS-15. For more details, see
https://github.com/actions/runner-images/issues/10721

franzpoeschel committed:
Properly check for empty HDF5_VERSION in CMake (#1702)

The old test apparently invoked some weird CMake legacy behavior.
This should make the pinned HDF5 Conda version unnecessary (#1701) since
the version with missing HDF5_VERSION is correctly dealt with.

franzpoeschel committed:
HDF5: Delete and re-create attribute when overwriting with diff. type (#1697)

* HDF5: Delete and re-create attribute when overwriting with diff. type

* WIP: Change datatype in test

Somehow truncation test is broken by this

* Add forgotten H5Aclose() call

franzpoeschel committed:
TOUCH IOTask: Avoid setting files as dirty in non-write modes (#1704)

* Bugfix: Consider dirty upon touch() only in write modes

* Add error check to avoid writing in read-only mode

franzpoeschel committed:
Changelog

franzpoeschel committed:
Use ULONG for particle patches

ax3l committed:
Update Changelog

ax3l committed:
Release Date
